### DISCUSSION TITLE
Multiline Lists

### DISCUSSION THREAD

**Discussion by HenningCode** [ratatui/ratatui#941] [Discussion]
Hello Guys,

whats the best way to go about multiline lists? 
I got some growing number of events I want to render with borders and the event name, time and description each on a new line.  I think this is not possible with List.

Can you guys give me a rough approach?

**Comment by joshka** [comment]
Hey there, multi-line items are documented at https://docs.rs/ratatui/latest/ratatui/widgets/struct.ListItem.html. Is this what you're after? That isn't documented in `List`. Do you have any ideas about what you'd add to List to make this easier to find?

**Reply by HenningCode** [reply]
Oh that was an easy find if you know that its there. I though just because the Parameter of `ListItem` is Text that it could not be multilined. 

But its not quite what I am after. I would like to draw borders around every `ListItem` with eg. Paragraph. Is that possible? 

I think this is not possible after I read more of the docs. So the best way to do is this is probably to check how many events fit onto the screen and than split the screen and render the amount of event `Paragaphs` that fit?

**Reply by joshka** [reply]
The built-in list doesn't yet support this (It may do in the future though). There is a third party widget list crate that could possibly be what you need. Check the showcase page on the website for a link.

**Reply by HenningCode** [reply]
Thanks! That exactly what I am looking for!
